spring-test/src/test/java/org/springframework/test/util/TestSocketUtilsTests.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.test.util;
+
+import javax.net.ServerSocketFactory;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalStateException;
+
+/**
+ * Unit tests for {@link  TestSocketUtils}.
+ *
+ * @author Sam Brannen
+ * @author Gary Russell
+ * @author Chris Bono
+ */
+class TestSocketUtilsTests {
+
+	@Test
+	void findAvailableTcpPort() {
+		int port = TestSocketUtils.findAvailableTcpPort();
+		assertThat(port >= 1024).as("port [" + port + "] >= " + 1024).isTrue();
+		assertThat(port <= 65535).as("port [" + port + "] <= " + 65535).isTrue();
+	}
+
+	@Test
+	void findAvailableTcpPortWhenNoAvailablePortFoundInMaxAttempts()  {
+		try (MockedStatic<ServerSocketFactory> mockedServerSocketFactory = Mockito.mockStatic(ServerSocketFactory.class)) {
```

Comment:
I am expecting this to get balked at for a couple of reasons...
1. "controversial" inline mock maker 
2. mocking statics in general

On **1)**, I am not overly concerned w/ using the [inline mock maker](https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#39) as I do not see any of the scenarios in `spring-test` that is has known issues with. Albeit, it might be considered a bit intrusive for only adding a single test case. 

On **2)**, I always have mixed feelings on mocking statics.

I figured I would throw this out here and see what the team suggests.

Reply:
Indeed, we don't have any plans to mock statics within Framework.

I have therefore revised the approach in a local commit that I will push as a follow up to merging your PR.